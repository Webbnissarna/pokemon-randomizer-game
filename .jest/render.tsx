import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
