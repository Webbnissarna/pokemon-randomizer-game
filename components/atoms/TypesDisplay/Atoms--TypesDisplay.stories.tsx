import { Flex } from "theme-ui";
import TypesDisplay from ".";
import type { Type } from "../../../utils/Generator/pokemon.types";

interface ControlProps {
  types: Type[];
}

export function Default({ types }: ControlProps) {
  return (
    <Flex sx={{ width: "min(200px, 100%)" }}>
      <TypesDisplay types={types} />
    </Flex>
  );
}

Default.args = {
  types: ["Electric", "Flying"],
} as ControlProps;
