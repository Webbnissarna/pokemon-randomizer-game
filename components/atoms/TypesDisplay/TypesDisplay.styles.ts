import type { ThemeUIStyleObject } from "theme-ui";

export const containerStyle: ThemeUIStyleObject = {
  width: "100%",
  justifyContent: "space-around",
  gap: "sm",
};

export const typeBoxStyle: ThemeUIStyleObject = {
  borderRadius: "sm",
  paddingX: "md",
  flex: 1,
  textAlign: "center",
};

export const textStyle: ThemeUIStyleObject = { fontSize: "sm" };
