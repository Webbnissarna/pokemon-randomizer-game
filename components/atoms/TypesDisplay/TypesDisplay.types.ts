import type { Type } from "../../../utils/Generator/pokemon.types";

export interface TypesDisplayProps {
  types: Type[];
}
