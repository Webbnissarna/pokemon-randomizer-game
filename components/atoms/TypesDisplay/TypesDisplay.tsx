import { Flex, Box, Text } from "theme-ui";

import { containerStyle, textStyle, typeBoxStyle } from "./TypesDisplay.styles";
import { TypeColors } from "../../../utils/pokemonMeta";

import type { TypesDisplayProps } from "./TypesDisplay.types";

export default function TypesDisplay({
  types,
}: TypesDisplayProps): JSX.Element {
  return (
    <Flex sx={containerStyle}>
      {types.map((type) => (
        <Box
          key={type}
          sx={{ ...typeBoxStyle, backgroundColor: TypeColors[type] }}
        >
          <Text sx={textStyle}>{type}</Text>
        </Box>
      ))}
    </Flex>
  );
}
