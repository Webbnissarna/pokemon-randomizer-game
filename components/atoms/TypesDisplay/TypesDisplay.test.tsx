import { render } from "../../../.jest/render";
import TypesDisplay from ".";

describe("TypesDisplay", () => {
  it("renders", () => {
    const { asFragment } = render(<TypesDisplay types={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
