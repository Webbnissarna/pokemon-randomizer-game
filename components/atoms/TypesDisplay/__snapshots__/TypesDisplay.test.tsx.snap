// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypesDisplay renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-j0o3ga-Box"
  />
</DocumentFragment>
`;
